### PR TITLE
Enable Alluxio to run as non-root in Kubernetes cluster

### PR DESCRIPTION
**Pull request by jiacheliu3**
This change will enable Alluxio to run as non-root user in Kubernetes cluster. 

The UFS local journal directory and the domain socket directory will have to be `chown` to `alluxio:alluxio`. This is done by adding `initContainers` which are executed before the containers. The `securityContext` are split up and moved to each container to provider security configuration in finer granularity.

Ref: [Kubernetes security context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/)

### CODE REVIEW

File: integration/kubernetes/helm/alluxio/templates/alluxio-master.yaml [hunk 1]
```diff
@@ -70,9 +72,16 @@ spec:
         app: {{ $masterName }}
     spec:
       securityContext:
-        runAsUser: {{ $root.Values.user }}
-        runAsGroup: {{ $root.Values.group }}
         fsGroup: {{ $root.Values.fsGroup }}
+      {{- if and $isSingleUfsLocal (not $isRoot) }}
+      initContainers:
+      - name: journal-chown
+        image: busybox
```
Comment (madanadit) [inline integration/kubernetes/helm/alluxio/templates/alluxio-master.yaml]: we should not create a dependency on a new image. modify the Dockerfile if needed to install `sudo` by `apk --no-cache --update add sudo`

File: integration/kubernetes/helm/alluxio/templates/alluxio-worker.yaml [hunk 2]
```diff
@@ -26,9 +28,16 @@ spec:
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
       securityContext:
-        runAsUser: {{ .Values.user }}
-        runAsGroup: {{ .Values.group }}
         fsGroup: {{ .Values.fsGroup }}
+      {{- if not $isRoot }}
+      initContainers:
+        - name: socket-chown
+          image: busybox
```
Comment (madanadit) [inline integration/kubernetes/helm/alluxio/templates/alluxio-worker.yaml]: same here

File: integration/kubernetes/multiMaster-embeddedJournal/alluxio-worker.yaml.template [hunk 3]
```diff
@@ -11,6 +11,7 @@
 # See the NOTICE file distributed with this work for information regarding copyright ownership.
 #
 
+
```
Comment (madanadit) [inline integration/kubernetes/multiMaster-embeddedJournal/alluxio-worker.yaml.template]: get rid of this newline

File: integration/kubernetes/helm/alluxio/templates/alluxio-master.yaml (hunk 1 above)
Comment (jiacheliu3) [reply integration/kubernetes/helm/alluxio/templates/alluxio-master.yaml]: Turned out I cannot `sudo` in initContainer (maybe not even possible in container commands in general). So I had to specify root in initContainer as following.
```
securityContext:
  runAsUser: 0
```

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**